src/test/java/io/javalin/TestRequestCache.java

```diff
@@ -0,0 +1,83 @@
+package io.javalin;
+
+import com.mashape.unirest.http.HttpResponse;
+import com.mashape.unirest.http.Unirest;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.entity.ByteArrayEntity;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.util.EntityUtils;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.io.InputStream;
+import java.util.Arrays;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+
+public class TestRequestCache {
+
+    private static Javalin app;
+    private static Javalin appCacheDisabled;
+    private static String origin;
+    private static String originCacheDisabled;
+
+    @BeforeClass
+    public static void setUp() {
+        app = Javalin.create()
```

Comment:
Use `port(0)` (random port) in all tests, then get the port using `app.port()`. 

If you set a specific port you might experience issues with collisions at some point. I was having troubles with Travis earlier, so I had to make a commit to always use random ports: fd95bc1deaa9b31e88d8968417f272b80f44075a